lib/LedgerSMB/File.pm

```diff
@@ -306,7 +307,7 @@ sub DEMOLISH {
        unlink $self->{file_path} . '/' . $file;
    }
    closedir (TMP);
-   rmdir $self->{file_path};
+   return rmdir $self->{file_path};
```

Comment:
No offense taken.  I was asking on lsmb dev for guidance on this, I will certainly take it now.

All your mentioned changes are up for review, plus a few cousins that show a strong family resemblance..